src/UnityExtension/Assets/Editor/GitHub.Unity/UI/BranchesView.cs

```diff
@@ -452,6 +454,8 @@ private void OnCreateGUI()
                     var cannotCreate = selectedNode == null ||
                                        selectedNode.Type == NodeType.Folder ||
                                        newBranchName == null ||
+                                       newBranchName.StartsWith("/") ||
+                                       newBranchName.EndsWith("/") ||
```

Comment:
Oh, actually there's a bunch more validation we should be doing. Spaces need to be turned into `-`, and https://www.kernel.org/pub/software/scm/git/docs/git-check-ref-format.html has a bunch of rules:

- They can include slash / for hierarchical (directory) grouping, but no slash-separated component can begin with a dot . or end with the sequence .lock.
- They cannot have two consecutive dots .. anywhere.
- They cannot have ASCII control characters (i.e. bytes whose values are lower than \040, or \177 DEL), space, tilde ~, caret ^, or colon : anywhere.
- They cannot have question-mark ?, asterisk *, or open bracket [ anywhere.
- They cannot begin or end with a slash / or contain multiple consecutive slashes
- They cannot end with a dot ..
- They cannot contain a sequence @{.
- They cannot be the single character @.
- They cannot contain a \.